File: string/0925_long_pressed_name/0925_long_pressed_name.py
class Solution(object):
    def isLongPressedName(self, name, typed):
        i = 0
        for j in xrange(len(typed)):
            if i < len(name) and name[i] == typed[j]:
                i += 1
            elif j == 0 or typed[j] != typed[j-1]:
                return False
        return i == len(name)

    
name = "alexx"
typed = "aaleexxx"

print(Solution().isLongPressedName(name, typed))

File: string/0925_long_pressed_name/0925_long_pressed_name.go
package main

import "fmt"

func isLongPressedName(name string, typed string) bool {
	if name == typed {
		return true
	}

	nameSize := len(name)
	typedSize := len(typed)

	i, j := 0, 0

	for i < nameSize && j < typedSize {
		c := name[i]
		need, pressed := 0, 0

		for i < nameSize && name[i] == c {
			need++
			i++
		}

		for j < typedSize && typed[j] == c {
			pressed++
			j++
		}

		if pressed < need {
			return false
		}
	}

	return i == nameSize && j == typedSize
}

func main() {
	name := "alexx"
	typed := "alleexxx"
	fmt.Println(isLongPressedName(name, typed))
}
